state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -478,12 +506,19 @@ fn parse_commit_block_info(data: &abi::Token) -> Result<Vec<CommitBlockInfoV1>>
         .into());
     };
 
+    let mut result = vec![];
     for d in data {
-        match CommitBlockInfoV1::try_from(d) {
-            Ok(blk) => res.push(blk),
-            Err(e) => tracing::error!("failed to parse commit block info: {e}"),
-        }
+        let block_info = {
+            if use_new_format {
+                CommitBlockInfo::V2(v2::CommitBlockInfo::try_from(d)?)
```

Comment:
Here I was thinking that _maybe_ there could be some kind of a generic structure where the `CommitBlockInfo` could be built in a way that this could be written like `CommitBlockInfo::<V2>::try_from(d)?` - maybe?

I cannot figure out a concrete suggestion how the type would specifically look like in this case, but this is just an intuitive gut feeling.

Reply:
Good idea! I pushed a new commit that abstracts away the whole `enum`. So, it's now only used internally when creating the type.